crates/matrix-sdk-indexeddb/src/event_cache_store/mod.rs

```diff
@@ -287,10 +287,27 @@ impl_event_cache_store! {
         &self,
         linked_chunk_id: LinkedChunkId<'_>,
     ) -> Result<Vec<ChunkMetadata>, IndexeddbEventCacheStoreError> {
-        self.memory_store
-            .load_all_chunks_metadata(linked_chunk_id)
-            .await
-            .map_err(IndexeddbEventCacheStoreError::MemoryStore)
+        let linked_chunk_id = linked_chunk_id.to_owned();
+        let room_id = linked_chunk_id.room_id();
+
+        let transaction = self.transaction(
+            &[keys::LINKED_CHUNKS, keys::EVENTS, keys::GAPS],
+            IdbTransactionMode::Readwrite,
+        )?;
+
+        let mut raw_chunks = Vec::new();
+        let chunks = transaction.get_chunks_in_room(room_id).await?;
+        for chunk in chunks {
+            let chunk_id = ChunkIdentifier::new(chunk.identifier);
+            let num_items = transaction.get_events_count_by_chunk(room_id, &chunk_id).await?;
+            raw_chunks.push(ChunkMetadata {
+                num_items,
+                previous: chunk.previous.map(ChunkIdentifier::new),
+                identifier: ChunkIdentifier::new(chunk.identifier),
+                next: chunk.next.map(ChunkIdentifier::new),
+            });
```

Comment:
If we have thousands chunks, this isn't going to scale well. I mean, it's probably fine for the moment, but a big `TODO` + a `timer!` macro is necessary here.

I suggest to add logs as I did in https://github.com/matrix-org/matrix-rust-sdk/pull/5382 (see commits by commits) with the `timer!` log macro. It's going to be pretty helpful to find which method is slow.

Reply:
Ah, interesting - I didn't think about that - thanks! And will do!